common/static/common/js/discussion/discussion_router.js

```diff
@@ -18,9 +18,20 @@
             return child;
         };
 
+    function getSingleThreadRoute(commentable_id, thread_id) {
+        return commentable_id + "/threads/" + thread_id;
+    }
+
     if (typeof Backbone !== "undefined" && Backbone !== null) {
         this.DiscussionRouter = (function(_super) {
 
+            var allThreadsRoute = "",
+                singleThreadRoute = getSingleThreadRoute(":forum_name", ":thread_id"), // :forum_name/threads/:thread_id
+                routes = [];
```

Comment:
should be `routes = {};` if it's going to be an object with string keys.

Also, just FYI, once edX switches to Babel / ES6 you can write this as:
```
DiscussionRouter.prototype.routes = {
    [allThreadsRoute]: "allThreads",
    [singleThreadRoute]: "showThread",
};
```